processor/memorylimiterprocessor/memorylimiter_test.go

```diff
@@ -440,3 +109,221 @@ func TestNoDataLoss(t *testing.T) {
 	err = processor.Shutdown(context.Background())
 	require.NoError(t, err)
 }
+
+// TestMetricsMemoryPressureResponse manipulates results from querying memory and
+// check expected side effects.
+func TestMetricsMemoryPressureResponse(t *testing.T) {
+	md := pmetric.NewMetrics()
+	ctx := context.Background()
+
+	tests := []struct {
+		name     string
+		mlCfg    *Config
+		getMemFn func() (uint64, error)
+		hasError bool
```

Comment:
```suggestion
		expectError bool
```

Why don't we have a case where it's `true`?

Reply:
test updated